lib/streamlit/runtime/metrics_util.py

```diff
@@ -213,6 +212,9 @@ def to_microseconds(seconds):
 def gather_metrics(callable: F) -> F:
     @wraps(callable)
     def wrap(*args, **kwargs):
+        # Avoid circular import
+        from streamlit.runtime.scriptrunner import get_script_run_ctx
```

Comment:
nit: Could this be moved to be imported outside of the `wrap` function?

I doubt this really matters (translation: this is certainly premature optimization) since every subsequent import of `get_script_run_ctx` should essentially do a dict lookup, realize that the function has already been imported, then be a no-op from there, but it feels a bit better paying this cost only in the few `@gather_metrics` annotation calls rather than in every wrapped `st.*` function when doing so just requires moving the import up a scope.

Reply:
Yes - and in fact, it can be moved all the way up to the top of the file, because it's not triggering a circular import anymore. Huzzah!